### PR TITLE
Nz/3686 rmv admin op

### LINKED ISSUE #3686: [Design issue] Remove "Administrative Operations" from the dropdown in the non-personnel state costs
The Administrative Operations category is not needed for non-personnel state costs for an MMIS APD and can be removed from the dropdown. **This option should stay for HITECH**.

- [ ]  designs are created, taking into consideration validation, multiple FFY, multiple entries, long entries, weird entries, and/or the export view if applicable
- [ ] 508 considerations and/or recommendations are reviewed
- [ ] designs are shared
- [ ] designs are approved by product

### PR DESCRIPTION
**Pull request by Sun-Mountain**
Resolves #3686 

### Description
The Administrative Operations category is not needed for non-personnel state costs for an MMIS APD and can be removed from the dropdown. **This option should stay for HITECH.**

### Significant changes or possible side effects
This should not affect anything outside of the `NonPersonnelCostForm` Component

### Automated test cases written

| Given | When | Then | Type (jest, tap, cypress) |
| ----- | ---- | ---- | ------------------------- |
| Apd Type | HITECH | There should be 7 dropdown options that include `Administrative operations` | jest |
| Apd Type | MMIS | There should be 6 dropdown options that does _not_ include `Administrative operations` | jest |

### Steps to manually verify this change

The dropdown under `Activity > State Staff and Expenses > Add State Expense`.

If it's a `HITECH` APD should look like:
Screen Shot 2023-01-10 at 2 35 25 PM

If it's an `MMIS` APD should look like:
Screen Shot 2023-01-10 at 2 36 09 PM

### This pull request is ready to code review when

- [x] Automated tests are updated (and all tests are passing)
- [x] New automated test cases are documented above
- [x] Pull request has been labeled, if applicable with feature, content, bug,
      tests, refactor
- [ ] Associated OpenAPI documentation has been updated
- [ ] The experience passes a basic manual accessibility audit (keyboard nav,
      screenreader, text scaling) OR an exemption is documented

### This pull request is ready to test when

- [x] Code has been reviewed by someone other than the original author

### This pull request is ready to review when the QA has

- [x] Verified the functionality related to the change
- [ ] Verified that the change works with Narrator on Windows
- [ ] Verified that the change works with VoiceOver on Mac
- [ ] Verified all updated pages with the WAVE tool
- [ ] Verified tab and keyboard navigation functionality

### This pull request can be merged when

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/src/pages/apd/activities/state-costs/NonPersonnelCost/NonPersonnelCostForm.js [hunk 1]
```diff
@@ -35,6 +39,24 @@ const NonPersonnelCostForm = forwardRef(
...
+    useEffect(() => {
+      if (apdType === APD_TYPE.HITECH) {
+        let newCatList = ['Administrative operations', ...categoryList];
+        setCatList(newCatList);
+      }
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [apdType]);
+
     const onSubmit = e => {
       console.log('onSubmit called', e);
```
Comment (thetif) [inline web/src/pages/apd/activities/state-costs/NonPersonnelCost/NonPersonnelCostForm.js:61]: remove console log

**Review by thetif — APPROVED** [review]
Works great! Great work!

File: web/src/pages/apd/activities/state-costs/NonPersonnelCost/NonPersonnelCostForm.test.js [hunk 2]
```diff
@@ -145,3 +164,25 @@ describe('the ContractorResourceForm component', () => {
...
+    resetLDMocks();
+  });
+
+  it('shows "Administrative operations" on HITECH APD', async () => {
+    await setup({}, hitechApd);
+
+    expect(screen.getAllByRole('option').length).toBe(7);
+  });
+
+  it('shows "Administrative operations" on HITECH APD', async () => {
```
Comment (amyd11) [inline web/src/pages/apd/activities/state-costs/NonPersonnelCost/NonPersonnelCostForm.test.js:183]: **Typo:** I think you meant to change the name of this test to be for MMIS showing the default list of options

File: web/src/pages/apd/activities/state-costs/NonPersonnelCost/NonPersonnelCostForm.test.js [hunk 3]
```diff
@@ -145,3 +164,25 @@ describe('the ContractorResourceForm component', () => {
...
+  });
+
+  afterAll(() => {
+    resetLDMocks();
+  });
+
+  it('shows "Administrative operations" on HITECH APD', async () => {
+    await setup({}, hitechApd);
+
+    expect(screen.getAllByRole('option').length).toBe(7);
```
Comment (amyd11) [inline web/src/pages/apd/activities/state-costs/NonPersonnelCost/NonPersonnelCostForm.test.js:180]: **Question (non-blocking):** Is it possible to check the list to see if it contains "Administrative operations" as an option?

**Review by amyd11 — COMMENTED** [review]
Works as expected! I think we just need to update the typo in the test.

**Review by thetif — APPROVED** [review]

File: web/src/pages/apd/activities/state-costs/NonPersonnelCost/NonPersonnelCostForm.test.js [hunk 4]
```diff
@@ -178,11 +178,31 @@ describe('Selection removes "Administrative operations" for in MMIS apd', () =>
...
     expect(screen.getAllByRole('option').length).toBe(7);
+
+    const dropdown = screen.getByRole('combobox');
+
+    await waitFor(() => {
+      dropdown.click();
+    });
+
+    expect(
+      await screen.findByText('Administrative operations')
```
Comment (amyd11) [inline web/src/pages/apd/activities/state-costs/NonPersonnelCost/NonPersonnelCostForm.test.js:189]: Awesome, thanks for adding that! I think it helps make it clear what the test is specifically checking for

**Review by amyd11 — APPROVED** [review]
Looks good!

**Comment by mirano-darren** [comment]
Looks good to me!
| Given | When | Then | Result |
| ----- | ---- | ---- | ------------------------- |
| Other State Expenses dropdown | on a HITECH APD | There should be 7 dropdown options that include `Administrative operations` | ✅ |
| Other State Expenses dropdown | on a MMIS APD | There should be 6 dropdown options that does _not_ include `Administrative operations` | ✅ |

**Review by mirano-darren — APPROVED** [review]

**Review by tbolt — APPROVED** [review]
Nice work

**Review by jeromeleecms — APPROVED** [review]
LGTM! Thanks for updating the order alphabetically.

**Pull request merged**